src/services/userAuth.ts

```diff
@@ -0,0 +1,62 @@
+import Cookies from 'js-cookie'
+import axios from 'axios';
+import config from "../config";
+
+const api = axios.create({
+  baseURL : config.API_BASE_ROUTE
+});
+
+// Add a request interceptor
+api.interceptors.request.use(
+  (config) => {
+    const token = Cookies.get('accessToken');
+    if (token) {
+      config.headers.Authorization = `Bearer ${token}`;
+    }
+    return config;
+  },
+  (error) => Promise.reject(error)
+);
+
+
+export default api;
+
+api.interceptors.response.use(
+  (response) => response,
+  async (error) => {
+      const originalRequest = error.config;
+
+      // If the error status is 403, it means the token has expired and we need to refresh it
+      if (error.response.status === 403 && error.response.status === 422) {
```

Comment:
How is this ever gonna be true

Reply:
This logic is for the refresh token authentication. It should run after the access token has expired. The refresh token logic is not working. As discussed in the call we can implement the refresh token in later sprint.